[E-Documents Core] - XRechnung processing using new import process

AndriusAndrulevicius opened the pull request:
Implements structured format reader for processing XRechnung electronic documents (invoices and credit memos) based on UBL 2.1 standard.

Enables automatic parsing of XML content into draft purchase documents with support for:
- Vendor identification through VAT registration, GLN, or participant ID
- Document header information (dates, amounts, currency, references)
- Line item details (products, quantities, prices, VAT rates)
- Customer and supplier party information extraction

Provides viewing capability for imported documents through readable purchase document interface.

> [!IMPORTANT]
> This Pull request depends on   #32

Fixes #

AndriusAndrulevicius committed:
Adds XRechnung format support for e-document import

Implements structured format reader for processing XRechnung electronic documents (invoices and credit memos) based on UBL 2.1 standard.

Enables automatic parsing of XML content into draft purchase documents with support for:
- Vendor identification through VAT registration, GLN, or participant ID
- Document header information (dates, amounts, currency, references)
- Line item details (products, quantities, prices, VAT rates)
- Customer and supplier party information extraction

Provides viewing capability for imported documents through readable purchase document interface.

AndriusAndrulevicius committed:
Refactors XRechnung handler to improve maintainability

Replaces hardcoded string literals with named constants throughout the XRechnung document processing logic.

Consolidates duplicate constants and moves them to appropriate scopes to reduce code duplication and improve readability.

Updates XML documentation comments to provide clearer descriptions of method functionality and parameter usage.

GMatuleviciute commented on Apps/DE/EDocumentDE/app/src/XRechnung/EDocumentXRechnungHandler.Codeunit.al:
```diff
@@ -0,0 +1,321 @@
+        EDocumentXMLHelper: Codeunit "EDocument XML Helper";
+        EMTok: Label 'EM', Locked = true;
+        "0198Tok": Label '0198', Locked = true;
+        "9930Tok": Label '9930', Locked = true;
```
Is " really needed?

AndriusAndrulevicius replied:
It is needed if the name doesn't start with a letter.
Changed the name of the variables

GMatuleviciute commented on Apps/DE/EDocumentDE/app/src/XRechnung/EDocumentXRechnungHandler.Codeunit.al:
```diff
@@ -0,0 +1,321 @@
+            end;
+            VendorParticipantId := SchemeID + ':' + EndpointID;
+        end;
+        VATRegistrationNo := CopyStr(EDocumentPurchaseHeader."Vendor VAT Id", 1, 20);
```
CopyStr in no other localizations I think this was needed is this case is exceptional for German?

AndriusAndrulevicius replied:
all other localizations have CopyStr. But I've updated the code to get rid of hardcoded value

AndriusAndrulevicius committed:
Update Apps/DE/EDocumentDE/app/src/XRechnung/EDocumentXRechnungHandler.Codeunit.al

Co-authored-by: Grasiele Matuleviciute <131970463+GMatuleviciute@users.noreply.github.com>

AndriusAndrulevicius committed:
Update Apps/DE/EDocumentDE/app/src/XRechnung/EDocumentXRechnungHandler.Codeunit.al

Co-authored-by: Grasiele Matuleviciute <131970463+GMatuleviciute@users.noreply.github.com>

AndriusAndrulevicius committed:
Scopes helpers, clarifies tokens, fixes VAT len

Limits telemetry and import helper to local scope for better encapsulation.
Improves token naming for routing ID and VAT registration clarity.
Avoids hardcoded length by using field max length to prevent VAT ID truncation.

AndriusAndrulevicius committed:
Merge remote-tracking branch 'origin/main' into dev/aan/XRechnung-v2

AndriusAndrulevicius committed:
Add XRechnung structured import tests and mocks

- Introduces integration tests for XRechnung invoice ingestion: content extraction into draft, viewing extracted data, draft updates, and purchase invoice creation.
- Provides a mock e-document format and a test utility library to set up services, data, and mappings.
- Adds an XML test resource and configures the test app to load resources; extends the test object ID range.
- Adds a helper to reset e-document draft data.
- Updates dependencies to include the DE tests extension.

Improves confidence in end-to-end structured import and draft processing.

GMatuleviciute committed:
Update Apps/DE/EDocumentDE/test/src/EDocumentStructuredTests.Codeunit.al
